### PR TITLE
Update jsonrpc requests with protocol requirements

### PR DESCRIPTION
**Pull request by llbartekll**

### CODE REVIEW

**Comment by MisterVants** [comment]
I resolved the conflict from merging my types rename PR for you.

File: Sources/WalletConnect/Engine/SessionEngine.swift [hunk 1]
```diff
@@ -325,28 +324,29 @@ final class SessionEngine {
         relayer.respondSuccess(for: payload)
     }
     
-    private func wcSessionNotification(_ payload: WCRequestSubscriptionPayload, notificationParams: SessionType.EventParams) {
+    private func wcSessionNotification(_ payload: WCRequestSubscriptionPayload, eventParams: SessionType.EventParams) {
```
Comment (MisterVants) [inline Sources/WalletConnect/Engine/SessionEngine.swift:327]:
```suggestion
    private func wcSessionEvent(_ payload: WCRequestSubscriptionPayload, eventParams: SessionType.EventParams) {
```

File: Tests/IntegrationTests/ClientTest.swift [hunk 2]
```diff
@@ -291,7 +291,7 @@ final class ClientTests: XCTestCase {
         let uri = try! await proposer.client.connect(blockchains: [], methods: [], events: [])!
 
         try! responder.client.pair(uri: uri)
-        let notificationParams = Session.Event(type: "type1", data: AnyCodable("notification_data"))
+        let notificationParams = Session.Event(type: "type1", data: AnyCodable("notification_data"), chainId: nil)
```
Comment (MisterVants) [inline Tests/IntegrationTests/ClientTest.swift:294]:
```suggestion
        let eventParams = Session.Event(type: "type1", data: AnyCodable("notification_data"), chainId: nil)
```

File: Tests/IntegrationTests/ClientTest.swift [hunk 3]
```diff
@@ -311,7 +311,7 @@ final class ClientTests: XCTestCase {
         let uri = try! await proposer.client.connect(blockchains: [], methods: [], events: [])!
 
         try! responder.client.pair(uri: uri)
-        let notificationParams = Session.Event(type: "type2", data: AnyCodable("notification_data"))
+        let notificationParams = Session.Event(type: "type2", data: AnyCodable("notification_data"), chainId: nil)
```
Comment (MisterVants) [inline Tests/IntegrationTests/ClientTest.swift:314]:
```suggestion
        let eventParams = Session.Event(type: "type2", data: AnyCodable("notification_data"), chainId: nil)
```

File: Sources/WalletConnect/Engine/SessionEngine.swift [hunk 4]
```diff
@@ -188,7 +188,7 @@ final class SessionEngine {
             case .sessionUpdateExpiry(let updateExpiryParams):
                 wcSessionUpdateExpiry(subscriptionPayload, updateExpiryParams: updateExpiryParams)
             case .sessionEvent(let eventParams):
-                wcSessionNotification(subscriptionPayload, notificationParams: eventParams)
+                wcSessionNotification(subscriptionPayload, eventParams: eventParams)
```
Comment (MisterVants) [inline Sources/WalletConnect/Engine/SessionEngine.swift:191]:
```suggestion
                wcSessionEvent(subscriptionPayload, eventParams: eventParams)
```

**Review by MisterVants — APPROVED** [review]
Added some suggestions to replace "notification" terminology.

File: Sources/WalletConnect/Storage/SessionStorage.swift [hunk 5]
```diff
@@ -39,4 +40,11 @@ final class SessionStorage: WCSessionStorage {
     func delete(topic: String) {
         storage.delete(topic: topic)
     }
+    
+    func getAcknowledgedSessions() -> [WCSession] {
+        getAll().compactMap {
```
Comment (MisterVants) [inline Sources/WalletConnect/Storage/SessionStorage.swift:45]: I think `filter()` would be more fitting to use here, instead

**Review by MisterVants — APPROVED** [review]

File: Sources/WalletConnect/Storage/SessionStorage.swift (hunk 5 above)
Comment (llbartekll) [reply Sources/WalletConnect/Storage/SessionStorage.swift]: absolutely

**Pull request merged**